src/vanilla.ts

```diff
@@ -48,19 +60,15 @@ export type StateCreator<T extends State, CustomSetState = SetState<T>> = (
   api: StoreApi<T>
 ) => T
 
-export default function create<TState extends State>(
-  createState: StateCreator<TState>
-): StoreApi<TState> {
+export default function create<
+  TState extends State,
+  CustomSetState extends SetState<any> = SetState<TState>
```

Comment:
I'm not sure what this `any` would cause. It could cause runtime error without compile error, technically?
@TkDodo can you review this PR please?